apps/web/src/components/kanban-board/task-card-context-menu/task-card-context-menu-content.tsx

```diff
@@ -0,0 +1,272 @@
+import {
+  Calendar as CalendarIcon,
+  Copy,
+  Flag,
+  FlipHorizontal2,
+  ListTodo,
+  Tags,
+  User,
+} from "lucide-react";
+
+import useCreateTask from "@/hooks/mutations/task/use-create-task";
+import useUpdateTask from "@/hooks/mutations/task/use-update-task";
+import useGetProjects from "@/hooks/queries/project/use-get-projects";
+import useGetWorkspaceUsers from "@/hooks/queries/workspace-users/use-get-workspace-users";
+
+import type { taskInfoSchema } from "@/components/task/task-info";
+import { Calendar } from "@/components/ui/calendar";
+import {
+  ContextMenuCheckboxItem,
+  ContextMenuItem,
+  ContextMenuSub,
+  ContextMenuSubContent,
+  ContextMenuSubTrigger,
+} from "@/components/ui/context-menu";
+import { generateLink } from "@/lib/generate-link";
+import type { Task } from "@/types/project";
+import { toast } from "sonner";
+import type { z } from "zod";
+
+interface TaskCardContext {
+  worskpaceId: string;
+  projectId: string;
+}
+
+interface TaskCardContextMenuContentProps {
+  task: Task;
+  taskCardContext: TaskCardContext;
+}
+
+export default function TaskCardContextMenuContent({
+  task,
+  taskCardContext,
+}: TaskCardContextMenuContentProps) {
+  const { data: projects } = useGetProjects({
+    workspaceId: taskCardContext.worskpaceId,
+  });
+  const { data: workspaceUsers } = useGetWorkspaceUsers({
+    workspaceId: taskCardContext.worskpaceId,
+  });
+  const { mutateAsync: updateTask } = useUpdateTask();
+  const { mutateAsync: createTask } = useCreateTask();
+
+  const projectOptions = projects?.map((project) => {
+    return { label: project.name, value: project.id };
+  });
+
+  const statusOptions = [
+    {
+      label: "To Do",
+      value: "to-do",
+    },
+    {
+      label: "In Progress",
+      value: "in-progress",
+    },
+    {
+      label: "In Review",
+      value: "in-review",
+    },
+    {
+      label: "Done",
+      value: "done",
+    },
+  ];
+
+  const usersOptions = workspaceUsers?.map((user) => ({
+    label: user.userName ? user.userName : user.userEmail,
+    value: user.userEmail,
+  }));
+
+  function handleCopyTaskLink() {
+    const path = `/dashboard/workspace/${taskCardContext.worskpaceId}/project/${taskCardContext.projectId}/task/${task.id}`;
+    const taskLink = generateLink(path);
+
+    navigator.clipboard.writeText(taskLink);
+    toast.success("Task link copied!");
+  }
+
+  async function handleDuplicateTaskToAnotherProject(projectId: string) {
+    if (!task) return;
```

Comment:
Do we need to check if it has a task? Since this component can be rendered only if it has a task.